### DISCUSSION TITLE
What is a good pattern for setting up a plugin for semantic release using GitHub Actions and conventional commits?

### DISCUSSION THREAD

**Discussion by dmoonfire** [atom/atom#22120] [Q&A] [answered]
As Larry Wall says, a good programmer is lazy.

In this specific regard, I use [semantic release](https://semantic-release.gitbook.io/semantic-release/) along with [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.2/) for almost all my self-managed projects. That includes a few Atom plugins.

The question is, what is a best/solid/generic way of setting up GitHub Actions to use that so I can accept [Dependbot's security PRs](https://github.com/dmoonfire/spelling-manager-js/pull/14) and have it automatically kick off a release that publishes the plugin. (I already check it manually before merging the PR.)

Obviously, I would have to assume that [having APM use conventional commits](https://github.com/atom/apm/issues/829) is resolved, but I'm willing to create a patch for that.

**Comment by UziTech** [comment, marked as answer]
[git-menu](https://github.com/UziTech/git-menu) is my Atom package that is using semantic release and renovate for automatic releases and dependency updates/security fixes.

**Reply by dmoonfire** [reply]
Thank you! That was exactly what I needed.
